spec/requests/issues_controller_spec.rb

```diff
@@ -149,6 +148,44 @@
         ),
       )
     end
+
+    describe "group access" do
+      before do
+        SiteSetting.discourse_jira_allowed_groups = Group::AUTO_GROUPS[:moderators]
+        Group.refresh_automatic_groups!
+      end
+
+      describe "regular user with insufficient permissions" do
+        it "does not allow access" do
+          sign_in(user)
+
+          post "/jira/issues.json"
+          expect(response.status).to eq(403)
+        end
+      end
+
+      describe "user in group with permission" do
+        it "allows access" do
+          post = Fabricate(:post)
+          project = Fabricate(:jira_project, uid: 2)
+
+          sign_in(user)
+
+          mods = Group.find(Group::AUTO_GROUPS[:moderators])
+          mods.add(user)
+
+          expect do
+            post "/jira/issues.json",
+                 params: {
+                   project_id: project.id,
+                   issue_type_id: issue_type.id,
+                   topic_id: post.topic_id,
+                   post_number: post.post_number,
+                 }
+          end.not_to raise_error
```

Comment:
I think this could lead to some false positives. Could you check it against the actual response and its status?